File: codeforces/cf301-325/cf324/e.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

#define REP(i, a, b) for (int i = (int)(a); i < (int)(b); i++)
#define rep(i, a) REP(i, 0, a)
#define EACH(i, a) for (auto i: a)
#define ITR(x, a) for (__typeof(a.begin()) x = a.begin(); x != a.end(); x++)
#define ALL(a) (a.begin()), (a.end())


int n;
int p[2005], s[2005], pos[2005];
ll ans;
vector<pair<int, int>> swaps;

void dodo() {
  rep(i, n) REP(j, i + 1, n) {
    if ((pos[p[i]] >= j && j > i) && (pos[p[j]] <= i && i < j)) {
    swaps.push_back(make_pair(i + 1, j + 1));
    ans += j - i;
    swap(p[i], p[j]);
    }
  }
}


void rev_do() {
  for(int i = n - 1; i > -1; i--) for (int j = i - 1; j > -1; j--) {
    if ((pos[p[j]] >= i && i > j) && (pos[p[i]] <= j && j < i)) {
    swaps.push_back(make_pair(i + 1, j + 1));
    ans += i - j;
    swap(p[i], p[j]);
    }
  }
}

bool isSame(int p[], int s[]) {
    rep(i, n) if (p[i] != s[i]) return false;
    return true;
}

int main(void) {
  // breath deeply and calm down
  cin >> n;
  rep(i, n) cin >> p[i];
  rep(i, n) cin >> s[i];
  rep(i, n) pos[s[i]] = i;

  while (!isSame(p, s)) {
    dodo();
    rev_do();
  }

  cout << ans << endl;
  cout << swaps.size() << endl;
  for (auto ele: swaps) {
    cout << ele.first << " " << ele.second << endl;
  }

  return 0;
}

File: codeforces/cf301-325/cf324/e.py
def dodo():
    res = 0
    for i in xrange(n):
        if pos[p[i]] <= i: continue
        for j in xrange(i + 1, n):
            if pos[p[i]] >= j > i and pos[p[j]] <= i < j:
                res += j - i
                swap.append((i + 1, j + 1))
                p[i], p[j] = p[j], p[i]
    return res

def revdo():
    res = 0
    for i in xrange(n - 1, -1, -1):
        if pos[p[i]] >= i: continue
        for j in xrange(i - 1, -1, -1):
            if pos[p[j]] >= i > j and pos[p[i]] <= j < i:
                res += i - j
                swap.append((i + 1, j + 1))
                p[i], p[j] = p[j], p[i]
    return res

n = int(raw_input())
p = map(int, raw_input().split())
s = map(int, raw_input().split())
pos = [0] * (n + 1)
for i in xrange(n):
    pos[s[i]] = i

ans = 0
swap = []
while not p == s:
    ans += dodo()
    ans += revdo()

print ans
print len(swap)
for i, j in swap:
    print i, j

